main.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    (0..length - 1).each do |i|
+      yield(self[i])
+    end
+  end
+
+  def my_map(&proc)
+    temp = []
+    (0..length - 1).each do |i|
+      temp << [proc.call(self[i])] if proc
```

Comment:
- [ ] my_map returns a new array with the results of running block once for every element in enum.
- [ ] my_map returns an Enumerator if no block is given